### ISSUE TITLE
[FEAT]: Simultaneous Use of Multiple Beacon Nodes

### ISSUE THREAD

**Issue by olisovyi-everstake** [Consensys-Incorporated/teku#8609]
Description:
I propose adding the ability to specify multiple beacon node endpoints (comma-separated) for validator clients. Instead of waiting for a failover when one beacon node fails, the validator client should send all tasks (e.g., attestation, block proposals, published blocks, duties) to all specified nodes **simultaneously**, similar to:
- lighthouse `--proposer-nodes` [flag](https://lighthouse-book.sigmaprime.io/advanced-proposer-only.html#the-beacon-node)
- nimbus [roles](https://nimbus.guide/validator-client-options.html#beacon-node-roles)

Additionally it allows nimbus and lighthouse to choose best MEV payload from **all specified** beacon nodes, making `getHeader` requests.

Use Case:
By using all beacon nodes at once, validator clients can avoid downtime or interruptions caused by waiting for failover. Even if one node becomes unavailable, the tasks are still processed by the remaining nodes, providing maximum reliability.

Benefits:
- No Failover Delays: Tasks are sent to all nodes immediately, avoiding wait times for failover.
- Increased Redundancy: Multiple nodes work in parallel, in HA manner 
- Validator
    - higher attestation performance
    - higher MEV rewards
    - no missed block proposals

Thank you for considering this request.

**Comment by ank-everstake** [comment]
You can check-out good explanation from [Vero VC](https://github.com/serenita-org/vero/blob/master/docs/using_multiple_beacon_nodes.md#block-proposals).

**Comment by mehdi-aouadi** [comment]
Thank you for this proposal.
I'd like to clarify few things before diving deeper into this feature:
> I propose adding the ability to specify multiple beacon node endpoints (comma-separated) for validator clients. Instead of waiting for a failover when one beacon node fails, the validator client should send all tasks (e.g., attestation, block proposals, published blocks, duties) to all specified nodes simultaneously, similar to:
> 
> lighthouse --proposer-nodes [flag](https://lighthouse-book.sigmaprime.io/advanced-proposer-only.html#the-beacon-node)
> nimbus [roles](https://nimbus.guide/validator-client-options.html#beacon-node-roles)
> Additionally it allows nimbus and lighthouse to choose best MEV payload from all specified beacon nodes, making getHeader requests.

Based on LH and Nimbus documentations, there's no guarantee/mention of choosing the best MEV payload when using such setup. The main purpose of using multiple BNs is primarily minimising the BN attack vector since a block proposer can be known ahead of the block proposal time.
Teku offers the [Sentry Beacon Nodes features](https://docs.teku.consensys.io/how-to/use-sentry-nodes) and the possibility of using multiple BNs using the [`beacon-node-api-endpoints` option](https://docs.teku.consensys.io/reference/cli/subcommands/validator-client#beacon-node-api-endpoint-beacon-node-api-endpoints).

> Benefits:
> - No Failover Delays: Tasks are sent to all nodes immediately, avoiding wait times for failover.
> - Increased Redundancy: Multiple nodes work in parallel, in HA manner 
> - Validator
>     - higher attestation performance
>     - higher MEV rewards
>     - no missed block proposals

There are also drawbacks:
- requires more resources (x number of running BNs)
- more complex logic for responses handling

> You can check-out good explanation from [Vero VC](https://github.com/serenita-org/vero/blob/master/docs/using_multiple_beacon_nodes.md#block-proposals).

The requested feature is more in line with what Vero does: Add a responses selection logic based on the task being performed (comparing the attestations with the head events, selecting the best aggregates/block...)

TLDR; we will discuss the advantages / drawbacks of this feature and keep you posted.
